package.json

```diff
@@ -24,7 +24,13 @@
     "prepare-release": "./prepare-release.sh",
     "release": "./release.sh",
     "prune-arc-build": "node ./pruneArcBuild.js",
-    "postinstall": "npm run prune-arc-build"
+    "postinstall": "npm run prune-arc-build",
+    "verify.initialize": "npm explore @daostack/arc -- npm install openzeppelin-solidity && npm explore @daostack/arc/node_modules/@daostack/infra -- npm install openzeppelin-solidity && npm explore @daostack/arc-hive -- npm install openzeppelin-solidity",
```

Comment:
why we need to re deploy ? migration repo already deploy arc , openzeppelin ,infra, arc-hive

Reply:
Need to reinstall because truffle-flattener explictely requires that it be run from within a folder that contains the truffle project that contains the contract being flattened.  Otherwise truffle-flattener won't work.

npm will install whatever version the package says needs to be installed.  I"m not sure how this could not be aligned with what the migration package is expecting.

Reply:
@orenyodfat  OK, I think I see how there can be misalignment.  The only way I can see to fix this is to change, for example, `"openzeppelin-solidity": "^2.2.0"` to `"openzeppelin-solidity": "2.2.0"`  in the arc package.  This would avoid picking up 2.2.1 when the contracts were compiled with 2.2.0.

Reply:
which version it will install on `npm install openzeppelin-solidity` ?

Reply:
@orenyodfat 

> which version it will install on npm install openzeppelin-solidity ?

It depends on the project and what is the current version available.

According to the package.json files:

Arc: ^2.1.3
Infra: ^2.1.3
arc-hive: ^2.2.0

These are the versions being installed today.

But what if openzeppelin-solidity releases a new version and suddenly there is a 2.2.1, and suppose this happens _after_ the release of arc-hive.  So arc-hive will have been compiled with 2.2.0, but will be flattened with 2.2.1, which could prevent the verification from succeeding.

The only reasonable way to solve this, that I see, is to remove the '^' from the openzeppelin-solidity version specification in the package.json files.

Reply:
but the script does that without version .
how about edit the script to do `npm install openzeppelin-solidity 2.2.1` ?

Reply:
> but the script does that without version
> how about edit the script to do npm install openzeppelin-solidity 2.2.1 ?

How is the script supposed to know what is the correct version?  I could write some code to parse the three package.json files to obtain the version number for openzeppelin-solidity in each package, but that would not be robust.  It would fail to obtain the correct version in the following case:

1. Assume Arc has ^2.2.0 in package.json
2. Assume the current version of openzeppelin-solidity is 2.2.1.
3. The user runs `truffle compile` on Arc prior to creating an NPM package for Arc.

RESULT:
The contracts are compiled with 2.2.1 but the package.json remains unchanged, still referring to ^2.2.0.

Then when the verify script parses the package.json, it will get 2.2.0, which is the wrong version, and thus verification may fail.

I think it is the right thing for us to do in general, not just for verify, to be explicit in the Arc, etc package.json files about exactly what version the contracts are compiled with.